docs/src/tutorial/minimal_working_compute.jl

```diff
@@ -0,0 +1,266 @@
+# configuration data to the shader runs. Make sure that this structure
+# corresponds to what is declared in the shader.
+struct shader_push_consts
+    val::Float32
+    n::UInt32
+end
+
+# Specialization constants are similar to push constants, but less dynamic: You
+# can change them before compiling the shader for the pipeline, but not
+# dynamically. This may have performance benefits for "very static" values,
+# such as block sizes.
+struct shader_spec_consts
+    local_size_x::UInt32
+end
+
+# Run the `glslangValidator` program to compile the shader.
+using glslang_jll
+glslang = glslangValidator(bin -> open(`$bin -V --quiet --stdin -S comp -o /dev/stdout `, read = true, write = true))
+write(glslang, shader_code) # send the code to stdin
+close(glslang.in) # send EOF
+shader_bcode = collect(reinterpret(UInt32, read(glslang))) #read the compiled shader spir-v code
+close(glslang)
+@assert glslang.exitcode == 0
+
+# Make a shader module with the code
+shader = unwrap(create_shader_module(device, sizeof(UInt32) * length(shader_bcode), shader_bcode))
+
+# ## Assembling the pipeline
+#
+# Descriptor set layout describes how many buffers of what kind are going to be
+# used by the shader
+dsl = unwrap(
+    create_descriptor_set_layout(
+        device,
+        [DescriptorSetLayoutBinding(0, DESCRIPTOR_TYPE_STORAGE_BUFFER, 1, SHADER_STAGE_COMPUTE_BIT, Sampler[])],
+    ),
+)
+
+# Pipeline layout describes the descriptor set together with the location of
+# push constants
+pl = unwrap(
+    create_pipeline_layout(device, [dsl], [PushConstantRange(SHADER_STAGE_COMPUTE_BIT, 0, sizeof(shader_push_consts))]),
+)
+
+# Shader compilation can use "specialization constants" that get propagated
+# (and optimized) into the shader code. We use them to make the shader
+# workgroup size "dynamic" in the sense that the size (32) is not hardcoded in
+# GLSL, but instead taken from here.
+const_local_size_x = 32
+spec_consts = [shader_spec_consts(const_local_size_x)]
+
+# Create a pipeline that can run the shader code with the specified layout.
+p = first(
+    first(
+        unwrap(
+            create_compute_pipelines(
+                device,
+                [
+                    ComputePipelineCreateInfo(
+                        PipelineShaderStageCreateInfo(
+                            SHADER_STAGE_COMPUTE_BIT,
+                            shader,
+                            "main", # this needs to match the function name in the shader
+                            specialization_info = SpecializationInfo(
+                                [SpecializationMapEntry(0, 0, 4)],
+                                UInt64(4),
+                                Ptr{Nothing}(pointer(spec_consts)),
+                            ),
+                        ),
+                        pl,
+                        -1,
+                    ),
+                ],
+            ),
+        ),
+    ),
+)
```

Comment:
I think we should break this down a bit:

```suggestion
pipeline_info = ComputePipelineCreateInfo(
    PipelineShaderStageCreateInfo(
        SHADER_STAGE_COMPUTE_BIT,
        shader,
        "main", # this needs to match the function name in the shader
        specialization_info = SpecializationInfo(
            [SpecializationMapEntry(0, 0, 4)],
            UInt64(4),
            Ptr{Nothing}(pointer(spec_consts)),
        ),
    ),
    pl,
    -1,
)
ps, _ = unwrap(create_compute_pipelines(device, [pipeline_info]))
p = first(ps)
```